### PR TITLE
[AutoDiff] Add missing derivatives to FloatingPointDifferentiation.swift.gyb

### PR DESCRIPTION
**Pull request by philipturner**
Add missing derivatives of protocol methods and enforce consistent code style. 

Partially resolves SR-15796.

`FloatingPoint.addProduct` is disabled because of SR-15793. Also, some functions will not be implemented until https://github.com/apple/swift/pull/40986 is resolved.

### CODE REVIEW

**Comment by philipturner** [comment]
I tried rebasing my commits to override the pull request's commit history, but it didn't work. I'd rather not bloat Swift's pull request history with a history of pull request attempts. Should I proceed with the PR I have here?

**Comment by philipturner** [comment]
CC: @rxwei who is best suited to review so many changes?

**Comment by philipturner** [comment]
SR-15794

File: stdlib/public/Differentiation/FloatingPointDifferentiation.swift.gyb [hunk 1]
```diff
@@ -54,205 +48,246 @@ extension ${Self}: Differentiable {
...
+    _ rhs: ${Self}
+  ) -> (value: Void, pullback: (inout ${Self}) -> ${Self}) {
     lhs += rhs
     return ((), { v in v })
   }
 
-  ${Availability(bits)}
-  @inlinable // FIXME(sil-serialize-all)
-  @_transparent
+  @inlinable
```
Comment (CodaFi) [inline stdlib/public/Differentiation/FloatingPointDifferentiation.swift.gyb:125]: Why did the transparency attributes come off these declarations?

Comment (philipturner) [reply stdlib/public/Differentiation/FloatingPointDifferentiation.swift.gyb:125]: I don't know why it was there to begin with.

Comment (CodaFi) [reply stdlib/public/Differentiation/FloatingPointDifferentiation.swift.gyb:125]: `@_transparent` forces inlining to occur at `-Onone`. `@inlinable` merely makes the definition of the associated function available to the inline, it does not make it mandatory.

Comment (philipturner) [reply stdlib/public/Differentiation/FloatingPointDifferentiation.swift.gyb:125]: This file hasn't been well-maintained for 2 years. There was even a function with an incorrect derivative that slipped by tests. I see no good justification for keeping @_transparent on just a portion of these functions.

**Comment by philipturner** [comment]
I’m closing this because S4TF doesn’t need these functions. All it needs is the `Builtin.applyDerivative_arityX` style functions to create a pullback, like what enables my [Differentiation](https://github.com/philipturner/differentiation) package.

Also, I think it could be brought up in a better way. Perhaps tackle the issue of unexplained `@_transparent` modifiers in one PR, the incorrect derivative in another, and the refactoring in a third.

Please, anyone use the tests I drafted for reference in a future pull request. I am not going to delete this branch on my fork, so you can merge my commits to a local branch and build off of them.

**Pull request closed without merging**